src/sys/socket/sockopt.rs

```diff
@@ -1115,6 +1115,158 @@ where
     }
 }
 
+/// Set the Upper Layer Protocol (ULP) on the TCP socket.
+///
+/// For example, to enable the TLS ULP on a socket, the C function call would be:
+///
+/// ```c
+/// setsockopt(sock, SOL_TCP, TCP_ULP, "tls", sizeof("tls"));
+/// ```
+///
+/// ... and the `nix` equivalent is:
+///
+/// ```ignore,rust
+/// setsockopt(sock, TcpUlp::default(), b"tls\0");
+/// ```
+#[cfg(target_os = "linux")]
```

Comment:
The TCP_ULP and SOL_TCP symbols are defined on Android as well as Linux.  So here and elsewhere, you should do
```suggestion
#[cfg(linux_android)]
```
Those two symbols are also defined on Fuchsia, but stuff like `tls12_crypto_info_aes_gcm_128` isn't, so I guess you should leave Fuchsia out.

Reply:
The TLS ULP is useless without being able to do `TLS_TX` and `TLS_RX`, and I only added all this in libc for `linux`. Someone who uses Android should extend it to Android.

Reply:
A, but you're right that there's no harm in enabling other ULPs at least. Done.